retrieval_service/datastore/providers/neo4j_graph.py

```diff
@@ -227,30 +227,34 @@ async def amenities_search(
     ) -> tuple[list[dict], Optional[str]]:
         async with self.__driver.session() as session:
             # OPTIONAL ensures that all similar amenities are returned even if they lack a SIMILAR_TO relationship
-            # Limit retrieval to 2 target nodes related to the source node, linked by SIMILAR_TO relationship
-            # Lower case relationship due to graph generator output
+            # Retrieve up to 2 source nodes along with all nodes connected by the SIMILAR_TO relationship
+            # Use lowercase relationship names to align with the graph generator's output format
+            # Similarity scores are bounded between 0 and 1, scores closer to 1 indicating higher similarity
             result = await session.run(
                 """
-                CALL db.index.vector.queryNodes('amenity_embedding', $top_k, $query_embedding)
-                YIELD node AS sourceAmenity
+                CALL db.index.vector.queryNodes('amenity_embedding', 2, $query_embedding)
```

Comment:
Should we still support top_k instead of hard coding it to 2?

Reply:
Any top-k greater than 2 will result in an extensive node list that's unpractical to work with